feat : BossPattern

jsuk10 opened the pull request:
보스 패턴 만들었습니다.

총 네가지 패턴으로
1. 노래하기(소환)
2. 화내기(운석소환)
3. 모자 부메랑
4. 레이저로 긁기
가 있으며

노래하기의 _maxSpownCount는 적은 값으로 해야 무한 루프에 안빠집니다.
상속할떄 필요없는 변수를 제거해야 하는데 차후에 진행하겠습니다.

_player의 경우에는 PatternController스크립트에서 관리해주기 때문에 플레이어를 여기에 직접 넣어줘야 합니다.

jsuk10 committed:
feat : Bullet looks at the player

jsuk10 committed:
feat : Fire when the player is in range

jsuk10 committed:
chore : Make a prefabs & test

jsuk10 committed:
feat : Make IPattern, Pattern, PatternController

패턴인터페이스와 패턴 컨트롤러를 만들고 패턴 인터페이스를 상속하고 추상화 시킴.

jsuk10 committed:
chore : Make a Moon And Projectiles Prefabs

jsuk10 committed:
feat : Make a Projectile (Parent)

jsuk10 committed:
feat : Make Meteo

회전함, 생성될때 플레이어를 찾아서 위치를 변경해서 내려감

jsuk10 committed:
feat : Hat

모자를 만듬
모자는 아래로 내려갈때 플레이어를 찾고
맨밑으로 내려가면 위로 다시 올라감.
반복후 안보이게 설정

jsuk10 committed:
feat : Brass

닿으면 죽음
평소에는 숨겼다가 돌려서 할예정

jsuk10 committed:
chore : Remove Remark

주석제거 및 필요없는 변수 삭제

jsuk10 committed:
feat : Hat Pattern

모자 던지는 패턴 만들었음, hat에 bool이 수정이 안되어서 프로퍼티 만듬

jsuk10 committed:
feat : Slash Pattern

회전하면서 브레스를 뿜는 패턴을 만들었음.

jsuk10 committed:
feat : Turns destruction into invisible

jsuk10 committed:
feat: Change remark property own position

Delete remark Create property change own position

jsuk10 committed:
feat : Makes the random pattern work normally

jsuk10 committed:
feat : Add TargetPlayerPosition Bool value

for hat
불값이 켜지면 y축을 초기화 안함 (hat 때문에)

jsuk10 committed:
feat : SingPattern

비활성화 된 객체를 활성화 시킴
주의사항으로는 _masSpownCount를 높게 잡아버리면 무한루프에 빠짐...
 - 모든 객체가 활성화 되면 루프를 계속 돌기 때문

jsuk10 committed:
feat : MadPattern

운석 떨구는 패턴 현재 약 4개를 받았고 마지막 운석이 비활성화 될 때 까지 종료 안됨

Kwon770 reviewed (changes requested):
NullReferenceException 뜨는데 확인해주세요

jsuk10 committed:
fix : NullReferenceException

플레이어가 할당되지 않아 널포인트가 뜨는 경우가 발생하여 없을 경우 할당해주게 만듬.

jsuk10 committed:
feat : AvoidInfiniteLoop

특정 조건에 무한 반복되어 유니티가 멈춰버리는 현상이 발생하기 떄문에 그것을 방지해주는 코드를 작성함.

jsuk10 committed:
refactor : rename Brass to Breath

jsuk10 committed:
fix : NullReferenceException

NullReferenceException: Object reference not set to an instance of an object
Enemy.HatPattern+d_2.MoveNext () (at Assets/2.Scripts/Enemy/HatPattern.cs:24)

Fix

jsuk10 committed:
feat ; add resource & arrange folders

jsuk10 committed:
refactor : rename _spwonInterval to  _spwanInterval

jsuk10 committed:
fix : Match file and class name

jsuk10 committed:
feat : Array renaming and other operations

jsuk10 committed:
feat : Destroy to SetActive

jsuk10 committed:
feat : Using parent-child to eliminate contradictions
